amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/records/ExportRecordsTests.java

```diff
@@ -0,0 +1,221 @@
+ * This class contains tests for
+ * Export REST API of RM Site
+ *
+ * @author Shubham Jain
+ * @since 3.5.0
+ */
+public class ExportRecordsTests extends BaseRMRestTest
+{
+    private RecordCategory rootCategory;
+
+    private RecordCategoryChild recordFolder;
+
+    private Record record;
+
+    private List<Record> recordList = new ArrayList<>();
+
+    @Autowired
+    private ExportAPI exportAPI;
+
+    @Autowired
+    private RecordCategoriesAPI categoriesAPI;
+
+    @Autowired
+    private RecordsAPI recordsAPI;
+
+    @BeforeClass (alwaysRun = true)
+    public void ExportRecordsTestsBeforeClass()
+    {
+        STEP("Create root level category");
+        rootCategory = createRootCategory(getRandomName("Category"));
+
+        STEP("Create the record folder inside the rootCategory");
+        recordFolder = createRecordFolder(rootCategory.getId(), getRandomName("Folder"));
+
+    }
+
+    @AfterTest
+    public void ExportRecordsTestsAfterTest(){
+
+        STEP("Delete the created Records");
+        for (int i = 0; i < recordList.size(); i++)
+        {
+            recordsAPI.deleteRecord(getAdminUser().getUsername(), getAdminUser().getPassword(), recordList.get(i).getName(),
+                    rootCategory.getName(), recordFolder.getName());
+
+        }
+        recordList.clear();
+    }
+
+    @AfterClass
+    public void ExportRecordsTestsAfter()
+    {
+
+        STEP("Delete the created folder");
+        deleteRecordFolder(recordFolder.getId());
+
+        STEP("Delete the created rootCategory");
+        categoriesAPI.deleteCategory(getAdminUser().getUsername(), getAdminUser().getPassword(), rootCategory.getName());
+
+    }
+
+
+    @DataProvider (name = "CreateRecordsInRecordFolder")
+    public Object[][] getCreatedRecordsFromRecordFolder()
+    {
+        STEP("Created a records in RM site with size varying from 4 MB to 200 MB");
+        for(int i = 4; i<200; i++){
+            createRecord("Record_"+i+"MB",i*1000000);
+            i=i+20;
+        }
+
+        return recordList.stream()
+                          .map(record -> new Object[] { record })
+                          .toArray(Object[][]::new);
+    }
+
+
+    /**
+     * Given a Root Category
+     * And an open record folder
+     * When I upload a Record with size greater than 4 MB
+     * Then I should be able to export it using APT
+     */
+    @Test (description = "Testing the Export functionality on RM site suing API for records of size >4MB",
+            priority = 1,enabled = true)
+    @AlfrescoTest (jira = "APPS-986")
+    public void ExportRecord() throws Exception
+    {
+
+        STEP("Create an electronic record in RM site with size > 4MB");
+        createRecord("Record_5MB", 5000000);
+
+        STEP("Export the created record with size greater than 4 MB and verifying the expected response code");
+        exportAPI.exportRecord(getAdminUser().getUsername(), getAdminUser().getPassword(),
+                SC_OK, record.getName());
```

Comment:
Why you just pass record.getId() and change the exportRecord method. In this way you can get rid off final String recNodeRef = getNodeRefSpacesStore() + contentService.getNodeRef(user, password, RM_SITE_ID,
                recordName);

Reply:
record.getName() has been passed instead of RecordID as to ExportNode method as the same method is being used to export both Record and RecordFolder.